Library/Homebrew/formula_installer.rb

```diff
@@ -212,6 +213,10 @@ def install
       opoo "#{formula.full_name}: #{old_flag} was deprecated; using #{new_flag} instead!"
     end
 
+    invalid_flag_names.each do |flag|
+      opoo "#{formula.full_name}: #{flag} is invalid for this formula and will be ignored!"
```

Comment:
Actually, I think you're right about making it a warning for now and we could make it a hard failure later.